### DISCUSSION TITLE
Security risk

### DISCUSSION THREAD

**Discussion by KennyGoi** [invertase/react-native-firebase#5919] [Q&A] [answered]
**Affected files**:
@react-native-firebase/crashlytics/android/src/main/java/io/invertase/firebase/crashlytics/ReactNativeFirebaseCrashlyticsModule.java
`67: System.exit(0);`

**Risks**:
Invocation of an exit method would immediately shut down the application.

If an exit method can be invoked through some logical flow within the application, it will cause a prompt Denial-of-Service (DoS).

**Comment by andersonaddo** [comment, marked as answer]
If you look at the source code, you'll notice that this is intended
https://github.com/invertase/react-native-firebase/blob/6a1125cad1bb79a169ca073cb7ddc12b467cc7ef/packages/crashlytics/lib/handlers.js#L115

**Reply by mikehardy** [reply]
I always get a chuckle out of automated "vulnerability scanners". They've got no subtlety or underestanding - @KennyGoi - it's up to you to interpret the results in context prior to raising them
